Snowflake.Data.Tests/IntegrationTests/SFDbDataReaderIT.cs

```diff
@@ -174,6 +174,33 @@ public void TestGetDate(string inputTimeStr)
             testGetDateAndOrTime(inputTimeStr, null, SFDataType.DATE);
         }
 
+        [Test]
+        public void TestDateOutputFormat()
+        {
+            using (IDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString = ConnectionString;
+                conn.Open();
+
+                IDbCommand cmd = conn.CreateCommand();
+                cmd.CommandText = "alter session set DATE_OUTPUT_FORMAT='MM/DD/YYYY'";
+                cmd.ExecuteNonQuery();
+
+                cmd.CommandText = $"select TO_DATE('2013-05-17')";
+                IDataReader reader = cmd.ExecuteReader();
+
+                Assert.IsTrue(reader.Read());
+                Assert.AreEqual("05/17/2013", reader.GetString(0));
+
+                reader.Close();
+
+                // set format back to default to avoid impact other test cases
+                cmd.CommandText = "alter session set DATE_OUTPUT_FORMAT='YYYY-MM-DD'";
```

Comment:
I believe we should perform this in `finally` block. Otherwise, if assertion fails, dirty connection returns to the pool.

Reply:
We have other tests like this without this cleanup. Imho to avoid what you suggest we could disable pooling for this test. Still that would require making this test run non parallel as rest of the tests would be affected with pooling disabled.

Reply:
Done. Put the clean up into finally block, while I think we can keep parallel and pooling enabled with this test case. The parameter is set on session level and cleared before it back to the pool and can be reused by other test case.